addons/medical_statemachine/functions/fnc_localityChangedEH.sqf

```diff
@@ -19,15 +20,50 @@
 params ["_unit", "_isLocal"];
 TRACE_2("localityChangedEH",_unit,_isLocal);
 
-if (!alive _unit) exitWith {TRACE_1("dead", _this)};
-
-if (!_isLocal) then {
-    // If locality changed, broadcast the last medical state
-    _unit setVariable [VAR_HEART_RATE, GET_HEART_RATE(_unit), true];
-    _unit setVariable [VAR_BLOOD_PRESS, _unit getVariable [VAR_BLOOD_PRESS, [80, 120]], true];
-    _unit setVariable [VAR_BLOOD_VOL, GET_BLOOD_VOLUME(_unit), true];
+if (!alive _unit) exitWith {};
 
+if (_isLocal) then {
     private _currentState = [_unit, EGVAR(medical,STATE_MACHINE)] call CBA_statemachine_fnc_getCurrentState;
-    TRACE_2("sending current state",_unit,_currentState);
-    [QGVAR(localityTransfer), [_unit, _currentState], _unit] call CBA_fnc_targetEvent;
+    TRACE_1("local",_currentState);
+
+    switch (true) do {
+        case (IN_CRDC_ARRST(_unit)): {
+            if (_currentState == "CardiacArrest") exitWith {};
+            _unit setVariable [VAR_CRDC_ARRST, false]; // force reset vars so setCardiacArrest can run (enteredStateCardiacArrest will also be called)
+            _unit setVariable [VAR_UNCON, false];
+            TRACE_1("manually changing state to CardiacArrest",_currentState);
+            [_unit, EGVAR(medical,STATE_MACHINE), _currentState, "CardiacArrest", {}, "LocalityChange"] call CBA_statemachine_fnc_manualTransition;
+        };
+        case (IS_UNCONSCIOUS(_unit)): {
+            if (_currentState == "Unconscious") exitWith {};
+            _unit setVariable [VAR_UNCON, false]; // force reset var so ace_medical_status_fnc_setUnconscious can run
+            TRACE_1("manually changing state to Unconscious",_currentState);
+            [_unit, EGVAR(medical,STATE_MACHINE), _currentState, "Unconscious", {}, "LocalityChange"] call CBA_statemachine_fnc_manualTransition;
+        };
+        case (IS_BLEEDING(_unit) || {IS_IN_PAIN(_unit)}): {
+            if (_currentState == "Injured") exitWith {};
+            TRACE_1("manually changing state to Injured",_currentState);
+            [_unit, EGVAR(medical,STATE_MACHINE), _currentState, "Injured", {}, "LocalityChange"] call CBA_statemachine_fnc_manualTransition;
+        };
+        default {
+            // If locality transfers back and forth, we could be in an old state and should transfer back to default
+            if (_currentState == "Default") exitWith {};
+            TRACE_1("manually changing state to Default",_currentState);
+            [_unit, EGVAR(medical,STATE_MACHINE), _currentState, "Default", {}, "LocalityChange"] call CBA_statemachine_fnc_manualTransition;
+        };
+    };
+} else {
+    /*
+    // Not sure if this is even needed, idea is that on locality transfer we broadcast more up to date info
```

Comment:
I think this should be the case in order to send the latest information, specially if the medical states vary between clients. In any case, it should be done before doing a manual transition to a particular state in my opinion.